float64 theta_old
float64 thetad_old
float64 theta
float64 thetad
float64 etheta
